fs/ensure_dir.ts

```diff
@@ -14,6 +14,24 @@ import { getFileInfoType } from "./_get_file_info_type.ts";
  * ```
  */
 export async function ensureDir(dir: string | URL) {
+  try {
+    const fileInfo = await Deno.lstat(dir);
```

Comment:
Using `lstat` instead of `stat` means that it will consider links to directories as links and not directories. I dont think it's good behavior as I often use links to trick programs into changing where files are stored. Any use case involving `ensureDir` will likely consider a link to a directory as acceptable.

Reply:
I've opened a PR to your branch here: https://github.com/kt3k/deno_std/pull/1

Reply:
That looks like a breaking change. I think we should work on that as an independent issue.